### PR TITLE
Theme Discovery: Implement LiTS Collection View

### PR DESCRIPTION
**Pull request by rcrdortiz**
Related to:
- https://github.com/Automattic/dotcom-forge/issues/4354

We need to implement the LiTS Collection view as specified in the parent issue so that we can release the Theme Discovery project to our users.

## Proposed Changes

* Implemented the designs needed.

Before:
Captura de pantalla 2023-11-07 a las 14 31 56

After:
Captura de pantalla 2023-11-07 a las 14 30 48

## Testing Instructions

* Open the Calypso live link
* Go to the LiTS and append `flags=themes/discovery-lits`
* Open the collection view of the Partner collection carousel
* You should see the updated style.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by dsas — COMMENTED** [review]
It works as described, there's just one small nitpick (which might be out of scope). 

It feels a bit off that none of the fonts in the heading and navigation match.

Collection view:
Screenshot 2023-11-07 at 14 31 19

Recommended view:
Screenshot 2023-11-07 at 14 31 30

The Collection 'Partner Themes' header is 30px, the Recommended 'Themes' header is 20px.

The Collection breadcrumbs don't match the Recommended page font size either, which might have been a different way to blend the two. The breadcrumbs include 'Themes' and clicking on that takes you back to Recommended where the 'Themes' header appears in nearly the same place but the collection view uses 14px and a different font.

**Comment by Copons** [comment]
Thanks @rcrdortiz!
Before looking at the code, there are a few design details to address. 🙂 

Screenshot 2023-11-07 at 14 24 17

On medium screen sizes, the collection is misaligned with the rest of the list. It seems that the culprit is the negative margin of `.themes-list .theme-collection__container`, which, at least on LiTS, should apparently be `-16px` instead of `-32px`.

I've noticed the same thing on LoTS too, but I think that, taking into account the variable-width sidebar, the numbers might end up different. Just be careful to not introduce regressions.

Screenshot 2023-11-07 at 14 17 56

On smaller sizes (I think between [sidebar disappears] and [font size becomes small]), the collection gains some small margins. It's not quite visible in the screenshot, but there's a `8px` whitespace on the collection sides.
This seem to only happen on LoTS.

Screenshot 2023-11-07 at 14 19 38

On LiTS, I think we should reuse the existing `NavigationHeader` component for the whole heading as we do [for the Showcase itself](https://github.com/Automattic/wp-calypso/blob/8d5c4b11318a44b6aa0d60dac7b3def449079ab9/client/my-sites/themes/theme-showcase-header.jsx#L76-L91).
If `NavigationHeader` is more involved than strictly needed, I think we can directly use `Breadcrumb` and `FormattedHeader` instead.

**Comment by rcrdortiz** [comment]
> On LiTS, I think we should reuse the existing NavigationHeader component for the whole heading as we do [for the Showcase itself](https://github.com/Automattic/wp-calypso/blob/8d5c4b11318a44b6aa0d60dac7b3def449079ab9/client/my-sites/themes/theme-showcase-header.jsx#L76-L91).
If NavigationHeader is more involved than strictly needed, I think we can directly use Breadcrumb and FormattedHeader instead.

Sure I'll change it to your suggestion.

>On medium screen sizes, the collection is misaligned with the rest of the list. It seems that the culprit is the negative margin of .themes-list .theme-collection__container, which, at least on LiTS, should apparently be -16px instead of -32px.

>I've noticed the same thing on LoTS too, but I think that, taking into account the variable-width sidebar, the numbers might end up different. Just be careful to not introduce regressions.

>Screenshot 2023-11-07 at 14 17 56
>On smaller sizes (I think between [sidebar disappears] and [font size becomes small]), the collection gains some small margins. It's not quite visible in the screenshot, but there's a 8px whitespace on the collection sides.
This seem to only happen on LoTS.

I'll create a new issue for these things since they're out of scope. The scope of the current issue is to apply changes to the collection view, not the Theme Showcase Collection Carousel.

**Comment by rcrdortiz** [comment]
> It works as described, there's just one small nitpick (which might be out of scope).
> 
> It feels a bit off that none of the fonts in the heading and navigation match.
> 
> Collection view: Screenshot 2023-11-07 at 14 31 19
> 
> Recommended view: Screenshot 2023-11-07 at 14 31 30
> 
> The Collection 'Partner Themes' header is 30px, the Recommended 'Themes' header is 20px.
> 
> The Collection breadcrumbs don't match the Recommended page font size either, which might have been a different way to blend the two. The breadcrumbs include 'Themes' and clicking on that takes you back to Recommended where the 'Themes' header appears in nearly the same place but the collection view uses 14px and a different font.

Not sure if they should match, maybe @lucasmendes-design has some input on this.

**Comment by rcrdortiz** [comment]
> On LiTS, I think we should reuse the existing NavigationHeader component for the whole heading as we do [for the Showcase itself](https://github.com/Automattic/wp-calypso/blob/8d5c4b11318a44b6aa0d60dac7b3def449079ab9/client/my-sites/themes/theme-showcase-header.jsx#L76-L91).
If NavigationHeader is more involved than strictly needed, I think we can directly use Breadcrumb and FormattedHeader instead.

@Copons for some reason the NavigationHeader component only shows the Title and Description [when there are less than 2 navigation items](https://github.com/Automattic/wp-calypso/blob/8d5c4b11318a44b6aa0d60dac7b3def449079ab9/client/components/navigation-header/index.tsx#L68). I'm not sure if it's a bug or a feature. I think we should keep the current implementation to avoid any unexpected regressions on other NavigationHeader consumers.

**Comment by Copons** [comment]
> @Copons for some reason the NavigationHeader component only shows the Title and Description [when there are less than 2 navigation items](https://github.com/Automattic/wp-calypso/blob/8d5c4b11318a44b6aa0d60dac7b3def449079ab9/client/components/navigation-header/index.tsx#L68). I'm not sure if it's a bug or a feature. I think we should keep the current implementation to avoid any unexpected regressions on other NavigationHeader consumers.

@rcrdortiz Yeah, that's fine. I was thinking of these kinds of strongly opinionated cases when I mentioned we could use the basic `Breadcrumb` and `FormattedHeader` components inside instead. 🙂

**Comment by lucasmendes-design** [comment]
> Not sure if they should match, maybe @lucasmendes-design has some input on this.


They should match :) We can use the same font sizes for both headers. The same spacing between the title and description as well. 

image

image

**Comment by rcrdortiz** [comment]
Updated with Dean's suggestion and also now I'm using the Breadcrumb and FormattedHeader components.

File: client/my-sites/themes/collections/theme-collection-view-header.scss [hunk 1]
```diff
@@ -3,7 +3,61 @@
...
+				padding-bottom: 22px;
+				margin: 16px;
+
+				@include break-small {
+					margin: 0;
+				}
+			}
+		}
+
+		&:not(.is-logged-in) {
```
Comment (Copons) [inline client/my-sites/themes/collections/theme-collection-view-header.scss:48]: We have a `.is-logged-out` selector.

**Review by Copons — APPROVED** [review]
This LGTM!

One minor change I'd suggest is to remove the `<p>` from the [description in the collection definition](https://github.com/Automattic/wp-calypso/blob/ea418d4d018a2460372681a7369c46d51a9f180b/client/my-sites/themes/collections/collection-definitions.tsx#L33), which would balance the space between heading and list:

| With `<p>` | Without `<p>` |
|--------|--------|
| Screenshot 2023-11-08 at 14 02 42 | Screenshot 2023-11-08 at 14 02 46 |

**Comment by rcrdortiz** [comment]
> This LGTM!
> 
> One minor change I'd suggest is to remove the `<p>` from the [description in the collection definition](https://github.com/Automattic/wp-calypso/blob/ea418d4d018a2460372681a7369c46d51a9f180b/client/my-sites/themes/collections/collection-definitions.tsx#L33), which would balance the space between heading and list:
> 
> With `<p>`	Without `<p>`
> Screenshot 2023-11-08 at 14 02 42	Screenshot 2023-11-08 at 14 02 46

Sure, I've removed it and it looks much better now.

**Pull request merged**